app/src/main/java/ar/com/wolox/android/bootstrap/ui/root/RootActivity.kt

```diff
@@ -1,22 +1,47 @@
 package ar.com.wolox.android.bootstrap.ui.root
 
+import android.os.Bundle
+import android.view.View
 import androidx.activity.viewModels
-import ar.com.wolox.android.bootstrap.ui.base.BaseActivity
-import ar.com.wolox.android.bootstrap.ui.login.LoginActivity
-import ar.com.wolox.android.bootstrap.ui.posts.PostsActivity
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import ar.com.wolox.android.bootstrap.databinding.ActivityBaseBinding
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
-class RootActivity : BaseActivity() {
+class RootActivity : AppCompatActivity() {
 
-    val viewModel: RootViewModel by viewModels()
+    private lateinit var binding: ActivityBaseBinding
 
-    override fun onResume() {
-        super.onResume()
-        if (viewModel.isUserLogged) {
-            PostsActivity.start(this)
-        } else {
-            LoginActivity.start(this)
-        }
+    private val viewModel: RootViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityBaseBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel.showLoadingLiveData.observe(
+            this,
+            Observer {
+                showLoading()
+            }
+        )
+
+        viewModel.hideLoadingLiveData.observe(
+            this,
+            Observer {
+                hideLoading()
+            }
+        )
```

Comment:
I'm not getting this implementation

Reply:
The shared viewmodel is used to communicate between fragments or between fragments and an activity in this case.
The activity observes the variable. If we want to call the function of the activity from the fragment we do that through the sharedviewmodel so that everything is decoupled.

Reply:
Nice!